### ISSUE TITLE
`require-module-attributes`: Support `TSImportType`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2726]
### Description

It seems typescript doesn't allow

```ts
type A = import("foo", {})
```

But allows

```ts
type A = import("foo", {with: {}})
```

I didn't add support for it in #2725, if someone need it, it can reuse the same logic as `ImportExpression`

### Examples

```ts
// ❌
type A = import("foo", {with: {}})

// ✅
type A = import("foo")
```

**Comment by sindresorhus** [comment]
I don’t think we should implement this.

TypeScript does support attributes on `import()` types, but only for `resolution-mode`. Empty TS import type attributes are already rejected by TypeScript (`TS1464`: exactly one key, `resolution-mode`, with value `import` or `require`), and `import("foo", {})` is a parse error.

So adding `TSImportType` handling here would add TypeScript-specific complexity just to lint code that cannot type-check.
